src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -115,11 +115,17 @@ def forward(self, x, position_ids, seq_len=None):
         # See https://github.com/huggingface/transformers/pull/29285
         device_type = x.device.type
         device_type = device_type if isinstance(device_type, str) else "cpu"
```

Comment:
```suggestion
        device_type = device_type if isinstance(device_type, str) and device_type != "mps" else "cpu"
```

Reply:
it seems to work well in my M1 Pro. I'll add a commit for this.